### DISCUSSION TITLE
Rewrite ITEM_BASIC

### DISCUSSION THREAD

**Discussion by daffa-darmawan** [forntoh/LcdMenu#124] [General]
do anyone have an example how to rewrite the ITEM_BASIC text?

**Comment by forntoh** [comment]
What do you mean by that?

**Reply by daffa-darmawan** [reply]
i want to change the text display on the lcd for the ITEM_BASIC type, is there any function to do it? also i found a bug where if you have two inputs on the same menu, the text that i input displayed on both of the inputs column.

**Reply by forntoh** [reply]
https://lcdmenu.forntoh.dev/usage

Do you want to change it when the code is running?

It will be impossible for me to help you if you're not more explicit, it's hard for me to get exactly what you want.

**Reply by melehin** [reply]
Hi

i think @daffa-darmawan wanted to change the text value of ITEM_BASIC
i did like this
```C
static char buf[10];
int value = 12345;
sprintf(buf, "%d", value);
menu[1]->setText(buf);
menu.update();
```
also i tried to show global variable based on mapper of ItemProgress. works fine too
```C
int variable = 10;
char* myMapper(uint16_t pos){
  static char buf[10];
  sprintf(buf, "%d", variable);
  return buf;
}
```
and usage
```C
variable = 10;
menu.update();
```

it would be nice to see the Progress and Command mixin.
to see a use case like this: "Run cmd and see progress"

also i want to see callback on enter() for submenu
